blocks/blocks.go

```diff
@@ -11,16 +12,20 @@ import (
 	mh "gx/ipfs/QmbZ6Cee2uHjG7hf19qLHppgKDRtaG4CVtMzdmK9VCVqLu/go-multihash"
 )
 
-var ErrWrongHash = errors.New("data did not match given hash!")
+// ErrWrongHash is returned when the Cid of a block is not the expected
+// according to the contents. It is currently used only when debugging.
```

Comment:
Ok, let's leave it as it is we might want to merge those errors in future. I must have not noticed it when I were writing the HashOnRead code.